src/Akeneo/Pim/Automation/DataQualityInsights/back/Domain/Model/ChannelLocaleRateCollection.php

```diff
@@ -49,14 +50,22 @@ function ($rawRate) {
         return $rateCollection;
     }
 
-    public static function fromNormalizedRates(array $normalizedRates, \Closure $getNormalizedRateValue): self
+    /**
+     * Format of a normalized of a Rate:
+     *  [
+     *      'rank'  => int, // Rank of the rate (from 1 to 5)
+     *      'value' => int, // Raw value (from 0 to 100)
+     *  ]
+     */
```

Comment:
We should use the array-shape definition to define the type and use static analysis of code covered by PHPStan: https://phpstan.org/writing-php-code/phpdoc-types#array-shapes


```suggestion
    /**
     * Format of a normalized of a Rate:
     *  [
     *      'rank'  => int, // Rank of the rate (from 1 to 5)
     *      'value' => int, // Raw value (from 0 to 100)
     *  ]
     * @param array{rank: int, value: int} $normalizedRates)
     */
```